services/app-api/src/resolvers/contract/withdrawAndReplaceRedundantRate.test.ts

```diff
@@ -0,0 +1,235 @@
+        const stateServer = await constructTestPostgresServer({
+            s3Client: mockS3,
+        })
+        const cmsServer = await constructTestPostgresServer({
+            context: {
+                user: cmsUser,
+            },
+            s3Client: mockS3,
+        })
+
+        const { contractID, replacementRateID, withdrawnRateID } =
+            await setupWithdrawRateTestData(stateServer)
+
+        const replaceReason = 'Admin has to do it!'
+        const cmsResult = await cmsServer.executeOperation({
+            query: WITHDRAW_REPLACE_RATE,
+            variables: {
+                input: {
+                    contractID,
+                    withdrawnRateID,
+                    replacementRateID,
+                    replaceReason,
+                },
+            },
+        })
+
+        const stateResult = await stateServer.executeOperation({
+            query: WITHDRAW_REPLACE_RATE,
+            variables: {
+                input: {
+                    contractID,
+                    withdrawnRateID,
+                    replacementRateID,
+                    replaceReason,
+                },
+            },
+        })
+
+        expect(cmsResult.errors).toBeDefined()
+        expect(stateResult.errors).toBeDefined()
+    })
+
+    it('returns error if the contract or individual rates are not currently submitted', async () => {
+        const stateServer = await constructTestPostgresServer({
+            s3Client: mockS3,
+        })
+        const cmsServer = await constructTestPostgresServer({
+            context: {
+                user: cmsUser,
+            },
+            s3Client: mockS3,
+        })
+        const adminServer = await constructTestPostgresServer({
+            context: {
+                user: adminUser,
+            },
+            s3Client: mockS3,
+        })
+
+        const contract1 = await createAndSubmitTestContractWithRate(stateServer)
+        const contract1ThatIsUnlocked = await unlockTestContract(
+            cmsServer,
+            contract1.id,
+            'Unlock reason'
+        )
+        const contractWithReplacementRate =
+            await createAndSubmitTestContractWithRate(stateServer)
+
+        const replaceReason = 'Admin has to do it!'
+        const adminResult = await adminServer.executeOperation({
+            query: WITHDRAW_REPLACE_RATE,
+            variables: {
+                input: {
+                    contractID: contract1ThatIsUnlocked.id,
+                    withdrawnRateID:
+                        contract1ThatIsUnlocked.packageSubmissions[0]
+                            .rateRevisions[0].rateID,
+                    replacementRateID:
+                        contractWithReplacementRate.packageSubmissions[0]
+                            .rateRevisions[0].rateID,
+                    replaceReason,
+                },
+            },
+        })
+        expect(adminResult.errors).toBeDefined()
+    })
```

Comment:
A couple more tests for some of the error cases would be nice. Like withdrawing a non-child rate or withdrawing an already withdrawn rate.

Reply:
See [45c04d0](https://github.com/Enterprise-CMCS/managed-care-review/pull/2590/commits/45c04d02fd5780108f414fca9b6fe1c51eb293e8)